src/widgetastic_patternfly/__init__.py

```diff
@@ -2351,3 +2351,111 @@ def item_select(self, item, close=True):
+            class item_list(ItemsList):
+                # define the item_class that the list-view uses
+                item_class = ItemClass
+            # define whatever else this page has on it ...
+
+    For an example: integration_tests/cfme/control/explorer/alerts.py::MonitorAlertsAllView
+
+    Usage is as follows assuming the list is instantiated as ``view.item_list``:
+
+    .. code-block:: python
+        # Access item by position
+        view.item_list[0].next() # => gives first item in list-view
+        # Iterate through rows
+        for item in view.item_list:
+            do_something()
+        # You can also filter items in two main ways
+        # 1) by an assoc_column that corresponds to an attribute of the item_class e.g.
+        item_list = view.item_list
+        item_list.assoc_column = 'description'
+        filtered_items = item_list[<desired_description>] # where <desired_description> is a str
+        # 2) by key-value pairs
+        item_filter = {'description': <desired_description>}
+        filtered_items = view.item_list[item_filter]
+        # note that filters can also be applied to the items() method
+        # e.g.
+        filtered_items = view.item_list.items(item_filter)
+
+    Args:
+         assoc_column: Name of an attribute/property defined in the item_class
+    """
+
+    ROOT = './/div[contains(@class,"list-view-pf-view")]'
+    ITEMS = './/div[contains(@class,"list-group-item-header")]'
+    item_class = None
+
+    def __init__(self, parent, assoc_column=None, logger=None):
+        View.__init__(self, parent, logger=logger)
+        self.assoc_column = assoc_column
+
+    def __getitem__(self, item_filter):
+        """ allows the ability to directly select AlertItem by a filter with
+            item = view.alerts_list[item_filter],
+            item_filter can be of type: string, dict, int, or None
+        """
+        return self.items(item_filter)
+
+    # properties
+    @property
+    def item_count(self):
+        """ returns how many rows are currently in the table."""
+        return len(self.browser.elements(self.ITEMS, parent=self))
+
+    # methods
+    def items(self, item_filter=None):
+        """ returns a generator for all Items matching the item_filter"""
+        start = 1
+        stop = self.item_count + 1
+        # filter via key, value pair
+        if isinstance(item_filter, dict):
+            key = item_filter.keys()[0]
+            value = item_filter.values()[0]
+            if len(item_filter) > 1:
+                self.logger.warning(
+                    "List-view filter currently not implemented for dictionaries"
+                    "greater than a length of one. Selecting the first key value pair from the dict"
+                )
+        # filter via string, note the default
+        elif isinstance(item_filter, str):
+            if self.assoc_column is None:
+                self.assoc_column = "description"  # default behavior
```

Comment:
I'm thinking this behavior should be in a property on the class, so that anywhere that we need to default assoc_column (future methods?)  it will have the same defaulting behavior.

In constructor, set `self._assoc_column`, then have the `assoc_column` property return `self._assoc_column or 'description'`

Reply:
Agreed, I will have include this in my updates